_posts/2019-09-23-kikcms.md

```diff
@@ -0,0 +1,157 @@
+* Pagination for large datasets
+* Add a record
+* Edit a record
+* Delete a record
+* Drag and drop rearranging
+* Functional checkboxes right in the table view
+* You can even have datatables in a datatable, as many levels as needed.
+* Form-fields like: Pick an image, autocomplete, rich-text, multi-checkbox, etc
+
+## Very powerful forms
+When you create a form, you can add fields to edit a table’s field. For example, the name of a person, but you can also easily edit related values in a form.
+
+Let’s consider this structure:
+
+![KikCMS example db structure](/assets/files/kikcms-db-structure.png "KikCMS example db structure")
+
+So you have a person, which has a name, a company, and several interests. To put this all in a single form, you must first define the relations, just like you are used to when using the Phalcon framework:
+
+```php
+public function initialize()
+{
+    parent::initialize();
+
+    $this->belongsTo('company_id', Company::class, 'id', ['alias' => 'company']);
+    $this->hasMany('id', PersonInterest::class, 'person_id', ['alias' => 'personInterests']);
+}
+```
+
+Now in the form, all you have to do is this:
+
+```php
+protected function initialize()
+{
+    $this->addTextField('name', 'Name');
+    $this->addTextField('company:name', 'Company name');
+    $this->addMultiCheckboxField('personInterests:interest_id', 'Interests', Interest::getNameMap());
+}
+```
+
+When adding a formfield, the first parameter is where it should be saved. In the case of a relation, you first specify the relation, followed by a ':' and then the field where it should be stored in the relation.
+
+Your form will look like this:
+
+![KikCMS example form](/assets/files/kikcms-form.png "KikCMS example form")
+
+Now when you press save, your data will be saved like this:
+
+test_person
+
+| id | name | company_id |
+|:------------- |:------------- |:------------- |
+| 1 | Tim Cook | 1 |
+
+test_company
+
+| id | name |
+|:------------- |:------------- |
+| 1 | Apple |
+
+test_person_interest
+
+| id | person_id | interest_id |
+|:------------- |:------------- |:------------- |
+| 1 | 1 | 1 |
+| 2 | 1 | 2 |
+| 3 | 1 | 3 |
+
+## Open source
+So the KikCMS has 3 main advantages: Usability, performance and rapid development. To me, these are very important. To give you an idea of what the KikCMS is capable of, this is what I’ve developed with it in the last 2 years:
+
+* 8 ‘regular’ websites with a few custom datatables
+* 4 webshops
+* A small SaaS for artists to display their art
+* A planning application for a tennis school
+* A platform for a community of freelancers, to showcase their portfolio, book rooms and buy tickets for events
+* A file database for a community
+* A SaaS application for weight loss consultants
+
+Now like I said, the KikCMS works very well for me. But I hope other people can benefit from it as well. That’s why I made it open source. Check it out here: https://github.com/krazzer/kikcms.
```

Comment:
KikCMS works perfectly for my needs. However, I made it open source, in order to give back to the community, and perhaps help others that have the same or similar requirements.

You can check KikCMS [here](https://github.com/krazzer/kikcms). Suggestions, comments and of course Pull requests are more than welcome! Do not forget to star the project on GitHub!

-- insert some social media accounts here? How can people find you etc. --

[krazzer](https://github.com/krazzer)

> If you are interested in having your project showcased in the Phalcon Blog, feel free to contact us at team@phalcon.io or reach out in our [Discord](https://phalcon.link/discord) server.
{: .alert .alert-info }